### PR TITLE
feat(mail): add template management shortcuts and --template-id flag

### PR DESCRIPTION
**Pull request by infeng**
## Summary
Adds email template management to lark-cli's `+mail` domain, rebased on top of latest `main`.

- New shortcuts: `mail +template-create`, `mail +template-update`
- New flag on existing shortcuts: `--template-id` on `+send` / `+draft-create` / `+reply` / `+reply-all` / `+forward`
- HTML inline image handling: scan `` local paths → Drive upload → rewrite as `cid:` refs, emit via `TemplateAttachment{id=file_key, is_inline=true, cid, ...}`
- Template does NOT carry `IsSendSeparately` — per-recipient separate-send is a draft-level concern and is intentionally not persisted on the template record.
- Template addresses use the IDL's native `mail_address` field only (no `address_alias` fallback).
- Key non-sensitive Info logs added at template-create / template-update / `--template-id` apply paths (mailbox_id, template_id, counts, name_len — never content / subject / recipient plaintext).

## Relationship to earlier PR
Supersedes the earlier branch `harness/01kpmfk36nw9fhn08mab2sbjkb` — that branch was based on an old `main` and mixed unrelated in-flight mail work. This branch rebuilds only the template work on top of current `upstream/main`.

## Test plan
- [ ] `go build ./...` passes locally
- [ ] `mail +template-create --help` surfaces the new flags
- [ ] `mail +send --template-id <id>` DryRun shows extra GET step + merged fields
- [ ] Inline image in `template_content` gets uploaded to Drive and CID-substituted

## Summary by CodeRabbit

* **New Features**
  * Add template support (--template-id) and explicit --subject to send, draft-create, reply, reply-all, forward; templates merge recipients/body/attachments with user flags, and user flags take precedence.
  * New shortcuts to create and update personal mail templates, including inline-image handling and attachment uploads.
  * Template large-attachment IDs are preserved and injected into outgoing drafts.

* **Documentation**
  * New docs for template workflows, cid image rewriting, dry-run behavior, permissions, and error guidance.

### CODE REVIEW

File: shortcuts/mail/template_compose.go [hunk 1]
```diff
@@ -0,0 +1,690 @@
...
+	// LARGE types; SMALL entries are reused by file_key server-side.
+	for _, att := range tpl.Attachments {
+		if att.ID == "" {
+			continue
+		}
+		res.LargeAttachmentIDs = append(res.LargeAttachmentIDs, att.ID)
+	}
+
+	return res
+}
```
[the remark this answers is not in the document]
Comment (infeng) [reply shortcuts/mail/template_compose.go:739]: Opted for the opposite direction: mail_reply.go no longer reads merged.Subject either. Reply/reply-all/forward keep the Re:/Fw: auto-subject (or user --subject) to preserve threading with the original conversation; applying the template subject would break that. Commit 8e3ad0d.

File: shortcuts/mail/mail_template_update.go [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+		newContent, _, err := resolveTemplateUpdateContent(runtime)
+		if err != nil {
+			return err
+		}
+		if newContent != "" {
+			tpl.TemplateContent = newContent
+		}
+		if runtime.Bool("set-plain-text") {
+			tpl.IsPlainTextMode = true
+		}
```
[the remark this answers is not in the document]
Comment (infeng) [reply shortcuts/mail/mail_template_update.go:151]: Wont-fix: `--patch-file` already provides the tri-state path (`{"is_plain_text_mode": false}` via JSON pointer). Promoting `--set-plain-text` to tri-state on the CLI layer would require either a string flag (`--set-plain-text=true|false`) or a dedicated `--unset-plain-text` companion, both of which surface more API surface for a rare use case (plain-text → HTML conversion of a template). Keeping the bool flag and directing users to `--patch-file` for the false case.

File: shortcuts/mail/mail_template_create.go [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+		return
+	}
+	if info.Size() <= common.MaxDriveMediaUploadSinglePartSize {
+		api.POST("/open-apis/drive/v1/medias/upload_all").Desc("Upload " + path)
+		return
+	}
+	api.POST("/open-apis/drive/v1/medias/upload_prepare").Desc("Large file prepare: " + path)
+	api.POST("/open-apis/drive/v1/medias/upload_part").Desc("Large file parts")
+	api.POST("/open-apis/drive/v1/medias/upload_finish").Desc("Large file finish")
+}
```
[the remark this answers is not in the document]
Comment (infeng) [reply shortcuts/mail/mail_template_create.go:201]: Wont-fix: the >3 GB case is extreme (MaxLargeAttachmentSize is 3 GB) and Execute path still enforces the cap via uploadToDriveForTemplate (template_compose.go:284). DryRun is a planning aid, not a pre-flight validator — users who exceed the cap get a real error the moment they drop --dry-run. Adding a duplicate size check here would drift from the similar existing shortcuts (draft-create / send) which also only enforce at Execute time.

**Comment by chanthuang** [comment]
Review findings:

1. `shortcuts/mail/mail_forward.go` 仍然在 `Validate` 阶段、模板合并之前做 `--confirm-send` 的收件人校验（大约在新增 `template-id` 校验后）。这会导致 `+forward --confirm-send --template-id <id>` 在模板是唯一收件人来源时被提前拒绝，根本走不到后面的 `applyTemplate()`。`+send` 已经为同类问题改成了 merge 后再校验，forward 这里也需要同样处理。

2. `shortcuts/mail/mail_forward.go` 在 `Execute` 末尾会把 `X-Lms-Large-Attachment-Ids` 写两次：一次写转发/用户新增的 LARGE 附件，一次再写模板带来的 LARGE 附件。`emlbuilder.Builder.Header()` 是追加语义，不是覆盖语义，所以一个同时包含两类 LARGE 附件的 forward 很容易只被服务端识别一部分。这里需要先合并两边的 attachment IDs，再只写一次 header。

3. `shortcuts/mail/mail_template_update.go` 当前的更新策略是“保留 GET 回来的全部 `tpl.Attachments`，再去重追加新上传的附件”，但不会清理新 `template_content` 已经不再引用的旧 inline 附件。结果就是用户每次替换/删除 `` 只能让附件列表越积越多，留下 orphaned 的 Drive-backed rows，最后把模板推向 `TemplateTotalSizeLimit`。如果新的 body 是 authoritative 的，这里在 full-replace PUT 之前应该先把不再被正文引用的 inline 附件过滤掉。

**Comment by infeng** [comment]
@chanthuang 感谢 review，三条都已在 642c921 修复：

1. **`mail_forward.go` `--confirm-send` 收件人校验时机**：参照 `+send` 的做法，当 `--template-id` 存在时 `Validate` 跳过 `validateComposeHasAtLeastOneRecipient`，改在 `Execute` 里 `applyTemplate` 合并完模板地址之后再做。这样 `+forward --confirm-send --template-id <id>` 在模板是唯一收件人来源时不会再被提前拒绝。

2. **`mail_forward.go` `X-Lms-Large-Attachment-Ids` 双写**：把 forward-derived（原邮件 LARGE + 用户 `--attach` 上传产生的 token）与 `templateLargeAttachmentIDs` 合并到一个 `[]largeAttID` 里、按 ID 去重，再做一次 `bld.Header(LargeAttachmentIDsHeader, base64(json))`，原来那条 `encodeTemplateLargeAttachmentHeader` 的二次写入直接删掉。现在不会再有两条同名 header。

3. **`mail_template_update.go` 孤儿 inline 附件**：在 `buildTemplatePayloadFromFlags` 拿到最终 rewritten content 之后、merge 新上传附件之前，新增一个仅在 `contentChanged` 时执行的过滤：丢弃 `IsInline=true && CID != ""` 但 `cid:<CID>` 在新 `template_content` 里不再出现的旧 attachment。non-inline 全部保留（无 cid: 引用关系），body 没改时整段跳过（避免把仍被旧内容引用的 inline 误删）。这样 `` 替换/删除不再向 attachments 里堆 orphan。

CodeRabbit 在 `mail_send / mail_reply / mail_reply_all / mail_draft_create` 也提了同款 dual-header 顾虑（其中前一条 review 评 Critical、后一条又改口说"only forward"），由于这次只是按你的 review 修，scope 内只动了 `mail_forward.go`；如果想一并对齐其它 3 个 compose 入口，告知一下我再补一个 commit。

**Comment by chanthuang** [comment]
Follow-up after `642c921`: the three functional issues from my previous review look fixed in the current diff.

One remaining request before merge: please add regression tests for these fixes. The fix commit changes behavior for three edge cases, but I do not see tests covering them yet:

- `+forward --confirm-send --template-id` where recipients come only from the template should pass validation after template merge, and a template that still has no recipients should fail.
- Forwarding with both forward/user LARGE attachments and template LARGE attachments should emit a single `X-Lms-Large-Attachment-Ids` header containing the merged IDs.
- `+template-update` should prune existing inline attachments whose CID is no longer referenced after `template_content` changes.

These are exactly the regression points we just found, so tests are the main remaining guardrail.

**Review by chanthuang — DISMISSED** [review]
lgtm

**Comment by infeng** [comment]
@chanthuang Added regression tests in `ae36b88e` covering all three points:

`shortcuts/mail/mail_cr_followup_test.go` — 7 tests, all passing locally:

1. **`+forward --confirm-send --template-id` recipient timing**
   - `TestMailForward_ConfirmSendTemplateOnlyRecipients_Allowed` — template-only recipients pass post-merge validation; the resulting EML contains the template-derived address.
   - `TestMailForward_ConfirmSendTemplateNoRecipients_Rejected` — template with no recipients fails with the `at least one recipient` error after merge (drafts.create stub deliberately omitted so `httpmock` would fail the test loudly if Execute slipped past the check).

2. **`+forward` single merged `X-Lms-Large-Attachment-Ids` header**
   - `TestMailForward_LargeAttachmentHeader_MergedSingleHeader` — source message with one LARGE attachment + template with two LARGE entries; asserts exactly **one** header line in the captured EML, and the base64-decoded JSON contains all three IDs (`src_large_1`, `tpl_large_a`, `tpl_large_b`).

3. **`+template-update` orphan inline prune**
   - `TestMailTemplateUpdate_OrphanInlineCIDPruned` — body change drops the `cid:` ref → orphan inline pruned from PUT.
   - `TestMailTemplateUpdate_StillReferencedInlineKept` — body change preserves the `cid:` ref → inline kept.
   - `TestMailTemplateUpdate_NonInlinePreservedOnContentChange` — non-inline rows always survive (no `cid:` to match against).
   - `TestMailTemplateUpdate_NoContentChangeKeepsAllInline` — body untouched (only `--set-subject`) → prune skipped, all existing inline rows kept.

`go test ./shortcuts/mail/...` 全部通过。

**Review by chanthuang — APPROVED** [review]
LGTM
